src/integrationTest/java/uk/gov/hmcts/reform/cwrdapi/FetchStaffProfileByIdIntegrationTest.java

```diff
@@ -0,0 +1,135 @@
+
+        Map<String, Object> response = (Map<String, Object>) caseworkerReferenceDataClient
+                .fetchStaffUserById(SearchStaffUserByIdResponse.class, path + searchString, ROLE_STAFF_ADMIN);
+        assertThat(response).containsEntry("http_status", HttpStatus.NOT_FOUND);
+        assertThat(((ErrorResponse)response.get("response_body")).getErrorDescription())
+                .contains(CaseWorkerConstants.NO_DATA_FOUND);
+    }
+
+
+    @Test
+    void should_return_status_code_404_when_profile_not_found_in_up() throws JsonProcessingException {
+        caseworkerReferenceDataClient.setBearerToken(EMPTY);
+        Map<String, Object> createResponse = caseworkerReferenceDataClient
+                .createCaseWorkerProfile(caseWorkersProfileCreationRequests, ROLE_CWD_ADMIN);
+        assertThat(createResponse).containsEntry("http_status", "201 CREATED");
+        String caseWorkerId = ((List)((Map)createResponse.get("body")).get("case_worker_ids")).get(0).toString();
+        Assertions.assertNotNull(caseWorkerId);
+
+        String path = "/profile/search-by-id";
+        String searchString = "?id=" + caseWorkerId;
+        caseworkerReferenceDataClient.setBearerToken(null);
+        userProfileGetUserByIdWireMock(caseWorkerId, 404);
+        Map<String, Object> getResponse = (Map<String, Object>) caseworkerReferenceDataClient
+                .fetchStaffUserById(SearchStaffUserByIdResponse.class, path + searchString, ROLE_STAFF_ADMIN);
+        assertThat(getResponse).containsEntry("http_status", HttpStatus.NOT_FOUND);
+        assertThat(((ErrorResponse)getResponse.get("response_body")).getErrorDescription())
+                .contains(CaseWorkerConstants.NO_DATA_FOUND);
+    }
+
+    @Test
+    void should_return_status_code_200_when_profile_found() throws JsonProcessingException {
+        Map<String, Object> createResponse = caseworkerReferenceDataClient
+                .createCaseWorkerProfile(caseWorkersProfileCreationRequests, ROLE_CWD_ADMIN);
+        assertThat(createResponse).containsEntry("http_status", "201 CREATED");
+        String caseWorkerId = ((List)((Map)createResponse.get("body")).get("case_worker_ids")).get(0).toString();
+        Assertions.assertNotNull(caseWorkerId);
+        String path = "/profile/search-by-id";
+        String searchString = "?id=" + caseWorkerId;
+        caseworkerReferenceDataClient.setBearerToken(null);
+
+        userProfileGetUserByIdWireMock(caseWorkerId, 200);
+        SearchStaffUserByIdResponse getResponse =  (SearchStaffUserByIdResponse)caseworkerReferenceDataClient
+                .fetchStaffUserById(SearchStaffUserByIdResponse.class, path + searchString, ROLE_STAFF_ADMIN);
+        Assertions.assertNotNull(getResponse);
+        Assertions.assertEquals(caseWorkerId, getResponse.getCaseWorkerId());
+        Assertions.assertNotNull(getResponse.getIdamStatus());
+        Assertions.assertEquals("pending", getResponse.getIdamStatus());
+
+    }
+
+}
```

Comment:
Functional test is missing for this new endpoint, as discussed pls add one

Reply:
As discussed with @kavithagade12 this functional test we are covering in the new ticket ... hence resolving this review comment as now will create new ticket  for this test ..fyi @kavithagade12